services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +43,56 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if (event == NULL){
+    return ERR_CODE_INVALID_ARG;
+  }
+
+  if (thermalMgrQueueHandle == NULL){
+    return ERR_CODE_INVALID_STATE;
+  }
+
+  if ((xQueueSend(thermalMgrQueueHandle, ( void * ) event, ( TickType_t ) 0)) != pdPASS){
+    return ERR_CODE_QUEUE_FULL;
+  }
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t interrupt;
+  interrupt.type = THERMAL_MGR_EVENT_INTERRUPT;
+
+  thermalMgrSendEvent(&interrupt);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  thermal_mgr_event_t event;
+  float temp;
+
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &event, 0) == pdPASS){
```

Comment:
I would set the block time to something larger than 0 since we don't have any reason to be polling the queue, it's better to stay blocked until there's something actually available in the queue